shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -544,7 +544,8 @@ - (BOOL)createShell:(NSString*)entrypoint
     if (!_platformViewsController) {
       _platformViewsController.reset(new flutter::FlutterPlatformViewsController());
     }
-    _publisher.reset([[FlutterObservatoryPublisher alloc] init]);
+    _publisher.reset([[FlutterObservatoryPublisher alloc]
```

Comment:
For a test, maybe put this in a separate method, OCMPartialMock it to return a OCMPartialMock of a publisher and make sure its enableObservatoryPublication is never called? We're already injecting ocmocked projects in flutterenginetests

Reply:
The hard part is mocking out the `flutter::DartServiceIsolate::AddServerStatusCallback`.  Let me see if I can figure it out.

Reply:
I got bogged down with this.  I want to merge this now to unblock g3 using this flag.
Filed https://github.com/flutter/flutter/issues/68221 to follow up.